src/map/pc_groups.cpp

```diff
@@ -3,462 +3,356 @@
-				    (inherit_count = config_setting_length(inherit)) <= 0) { // this group does not inherit from others
-					++i;
-					group_settings->inheritance_done = true;
-					continue;
-				}
-				
-				for (j = 0; j < inherit_count; ++j) {
-					GroupSettings *inherited_group = NULL;
-					const char *groupname = config_setting_get_string_elem(inherit, j);
-
-					if (groupname == NULL) {
-						ShowConfigWarning(inherit, "pc_groups:read_config: \"inherit\" array member #%d is not a name, removing...", j);
-						config_setting_remove_elem(inherit,j);
-						continue;
-					}
-					if ((inherited_group = name2group(groupname)) == NULL) {
-						ShowConfigWarning(inherit, "pc_groups:read_config: non-existent group name \"%s\", removing...", groupname);
-						config_setting_remove_elem(inherit,j);
-						continue;
-					}
-					if (!inherited_group->inheritance_done)
-						continue; // we need to do that group first
-
-					// Copy settings (commands/permissions) that are not defined yet
-					if (inherited_group->commands != NULL) {
-						int l = 0, commands_count = config_setting_length(inherited_group->commands);
-						for (l = 0; l < commands_count; ++l)
-							config_setting_copy(commands, config_setting_get_elem(inherited_group->commands, l));
-					}
+	}
 
-					if (inherited_group->permissions != NULL) {
-						int l = 0, permissions_count = config_setting_length(inherited_group->permissions);
-						for (l = 0; l < permissions_count; ++l)
-							config_setting_copy(permissions, config_setting_get_elem(inherited_group->permissions, l));
-					}
+	if( this->nodeExists( node, "Inherit" ) ){
+		const YAML::Node& inherits = node["Inherit"];
+
+		auto& inheritanceVector = this->inheritance[groupId];
+
+		for( const auto& it : inherits ){
+			std::string inherit = it.first.as<std::string>();
+			bool enable = it.second.as<bool>();
 
-					++done; // copied commands and permissions from one of inherited groups
+			util::tolower( inherit );
+
+			if( enable ){
+				if( !util::vector_exists( inheritanceVector, inherit ) ){
+					inheritanceVector.push_back( inherit );
 				}
-				
-				if (done == inherit_count) { // copied commands from all of inherited groups
-					++i;
-					group_settings->inheritance_done = true; // we're done with this group
+			}else{
+				if( !util::vector_exists( inheritanceVector, inherit ) ){
+					this->invalidWarning( it, "Trying to remove inheritance of non-inherited group %s\n", inherit.c_str() );
+					return 0;
 				}
+
+				util::vector_erase_if_exists( inheritanceVector, inherit );
 			}
-			dbi_destroy(iter);
+		}
+	}
+
+	if( !exists ){
+		this->put( groupId, group );
+	}
+
+	return 1;
+}
 
-			if (++loop > group_count) {
-				ShowWarning("pc_groups:read_config: Could not process inheritance rules, check your config '%s' for cycles...\n",
-				            config_filename);
-				break;
+void PlayerGroupDatabase::loadingFinished(){
+	static const int MAX_CYCLES = 10;
```

Comment:
Should we note in the docs that we only inherit up to 10 groups?

Reply:
Its only if those groups inherit in circles. Unless that happens it is not limited intentionally.